networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/gossip/subnets/NodeBasedStableSubnetSubscriberTest.java

```diff
@@ -13,51 +13,56 @@
 import static org.mockito.ArgumentMatchers.intThat;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 
-import java.util.Random;
+import java.util.Optional;
 import org.jetbrains.annotations.NotNull;
 import org.junit.jupiter.api.Test;
 import tech.pegasys.teku.infrastructure.unsigned.UInt64;
 import tech.pegasys.teku.networking.eth2.Eth2P2PNetwork;
 import tech.pegasys.teku.spec.Spec;
 import tech.pegasys.teku.spec.TestSpecFactory;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
 
-public class ValidatorBasedStableSubnetSubscriberTest {
+public class NodeBasedStableSubnetSubscriberTest {
   protected Spec spec = TestSpecFactory.createMinimalPhase0();
   private final Eth2P2PNetwork network = mock(Eth2P2PNetwork.class);
-  private final int attestatonSubnetCount = spec.getNetworkingConfig().getAttestationSubnetCount();
+  private final int attestationSubnetCount = spec.getNetworkingConfig().getAttestationSubnetCount();
+
+  final DataStructureUtil dataStructureUtil =
+      new DataStructureUtil(TestSpecFactory.createDefault());
 
   @Test
-  void shouldSubscribeToNoMoreThanAttestationSubnetCount() {
-    final ValidatorBasedStableSubnetSubscriber subscriber = createSubscriber(256);
-    subscriber.onSlot(UInt64.ONE, 2);
-    verify(network, times(attestatonSubnetCount))
-        .subscribeToAttestationSubnetId(intThat(i -> i >= 0 && i < attestatonSubnetCount));
-    for (int i = 0; i < attestatonSubnetCount; i++) {
-      verify(network).subscribeToAttestationSubnetId(eq(i));
-    }
+  void shouldSubscribeToExpectedSubnets() {
+    final NodeBasedStableSubnetSubscriber subscriber = createSubscriber();
+    subscriber.onSlot(UInt64.ZERO);
+    verify(network, times(spec.getNetworkingConfig().getSubnetsPerNode()))
+        .subscribeToAttestationSubnetId(intThat(i -> i >= 0 && i < attestationSubnetCount));
   }
 
   @Test
-  void shouldSubscribeToExpectedSubnetsWithNoMinimum() {
-    final ValidatorBasedStableSubnetSubscriber subscriber = createSubscriber(0);
-    subscriber.onSlot(UInt64.ONE, 10);
-    verify(network, times(10))
-        .subscribeToAttestationSubnetId(intThat(i -> i >= 0 && i < attestatonSubnetCount));
+  void shouldSubscribeToNewSubnetsAfterExpiration() {
+    final NodeBasedStableSubnetSubscriber subscriber = createSubscriber();
+    subscriber.onSlot(UInt64.ONE);
+    verify(network, times(spec.getNetworkingConfig().getSubnetsPerNode()))
+        .subscribeToAttestationSubnetId(intThat(i -> i >= 0 && i < attestationSubnetCount));
+    subscriber.onSlot(
+        UInt64.ONE.plus(
+            UInt64.valueOf(
+                (long) spec.getNetworkingConfig().getEpochsPerSubnetSubscription()
+                    * spec.getSlotsPerEpoch(UInt64.ONE))));
+    verify(network, times(spec.getNetworkingConfig().getSubnetsPerNode() * 2))
+        .subscribeToAttestationSubnetId(intThat(i -> i >= 0 && i < attestationSubnetCount));
```

Comment:
I'd also check that 2 lists doesn't match. (I think they could have 1 equal subnet but definitely not both)

Reply:
Done, added an assertion to check that we have at least one different subnet id after two rounds